software/libdvi/tmds_encode.c

```diff
@@ -86,7 +86,7 @@ void __not_in_flash_func(tmds_encode_data_channel_8bpp)(const uint32_t *pixbuf,
 	int require_lshift = configure_interp_for_addrgen(interp0_hw, channel_msb, channel_lsb, 0, 8, 6, tmds_table);
 	int lshift_upper = configure_interp_for_addrgen(interp1_hw, channel_msb, channel_lsb, 16, 8, 6, tmds_table);
 	assert(!lshift_upper); (void)lshift_upper;
-	if (require_lshift)	
+	if (require_lshift || (DVI_SYMBOLS_PER_WORD==1))
```

Comment:
@mlorenzati - Is adding `|| (DVI_SYMBOLS_PER_WORD==1)` the fix for RGB332, or is there more to it?

Reply:
As far as I remember, yes, I implemented it first on my pico-rgb2hdmi project
https://github.com/mlorenzati/pico-rgb2hdmi/commit/0863b7decedf62c720b7e2f2bf98000806d08149